api/sessionrecording/session_recording.go

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sessionrecording
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"io"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/defaults"
+	apievents "github.com/gravitational/teleport/api/types/events"
+)
+
+const (
+	// Int32Size is a constant for 32 bit integer byte size
+	Int32Size = 4
```

Comment:
Now that this is in API, we should try to export only what is absolutely necessary. If these values aren't needed outside the package, let's unexport them.

Reply:
I moved the `*Header` constants back to `lib/events` since `api/sessionrecording` isn't using them. I was using them in another repository, but not super concerned about not being able to import them directly, especially since this package doesn't need them.


---

What are your thoughts on these remaining constants? `Int32Size`, `Int64Size`, `MaxProtoMessageSizeBytes`, and `ProtoStreamV1` are used by both `api` and `lib`.

1. Is it cool to keep these exported to keep them in sync between the two packages?
1. Or do you prefer to keep these unexported to avoid users from relying on these? And both `lib` and `api` can both have their own respective constants.

I'm good either way. I was leaning towards the first choice, but can understand not wanting users to rely on these constants.